src/gwt/src/org/rstudio/studio/client/common/mathjax/MathJaxLoader.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * MathJaxLoader.java
+ *
+ * Copyright (C) 2009-16 by RStudio, Inc.
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+package org.rstudio.studio.client.common.mathjax;
+
+import com.google.gwt.core.client.Scheduler;
+import com.google.gwt.core.client.Scheduler.RepeatingCommand;
+import com.google.gwt.dom.client.Document;
+import com.google.gwt.dom.client.HeadElement;
+import com.google.gwt.dom.client.ScriptElement;
+import com.google.inject.Singleton;
+
+@Singleton
+public class MathJaxLoader
+{
+   public interface Callback
+   {
+      public void onLoaded(boolean alreadyLoaded);
+   }
+   
+   public MathJaxLoader()
+   {
+      ensureMathJaxLoaded();
+   }
+   
+   public static void ensureMathJaxLoaded()
+   {
+      ensureMathJaxLoaded(null);
+   }
+   
+   public static boolean isMathJaxLoaded()
+   {
+      return MATHJAX_LOADED;
+   }
+   
+   public static void ensureMathJaxLoaded(Callback callback)
+   {
+      if (MATHJAX_LOADED)
+      {
+         callback.onLoaded(MATHJAX_LOADED);
+         return;
+      }
+      
+      initializeMathJaxConfig();
+      ScriptElement mathJaxEl = createMathJaxScriptElement();
+      HeadElement headEl = Document.get().getHead();
+      headEl.appendChild(mathJaxEl);
+      Scheduler.get().scheduleFixedDelay(new RepeatingCommand()
```

Comment:
I gave this a shot, and it wasn't quite what we needed -- we still actually need to wait for MathJax itself to finish initialization, and AFAICS the best way to do that is to just poll on `MathJax.isReady`.